torch/nn/modules/module.py

```diff
@@ -2021,6 +2043,7 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
 
         def load(module, local_state_dict, prefix=''):
             local_metadata = {} if metadata is None else metadata.get(prefix[:-1], {})
+            local_metadata['assign_to_params_buffers'] = int(assign)
```

Comment:
Why convert it to an int?

Reply:
Using a bool here makes [this legacy jit test ](https://www.internalfb.com/code/fbsource/[5d916f94b781450e74bda5e477213319be1948cd]/fbcode/caffe2/test/jit/test_module_apis.py?lines=38) fail with  

```
RuntimeError: Tracer cannot infer type of {'version': 1, 'assign_to_params_buffers': False}
2023-05-24T21:55:39.2362236Z :Dictionary inputs to traced functions must have consistent type. Found int and bool
``` :/

Reply:
cc @davidberard98 do you know what this error is about? Are you ok with us silency this legacy test?

Reply:
It occurred to me that we can make this backward compatible by not adding assign to the dict if it is False. I have just made this change torch.jit.script will probably fail if someone uses it with load_state_dict assign=True but I suppose that is unlikely (?)

Reply:
fwiw, I think state dict is something that torchscript users care about so we shouldn't break it in the general case. But I think it's probably fine if assign=True doesn't work.

cc @IvanKobzarev who I think is working on state_dict torchscript stuff for torchrec purposes, to confirm that this isn't something that he cares about